src/api/api.ts

```diff
@@ -31,54 +81,244 @@ export class GhostApi {
+  }
+
+  public uploadRoutes(
+    getReadStream: () => fs.ReadStream
+  ): Promise<RoutesResponse> {
+    if (!this.token) {
+      throw new Error('Before uploading routes you must call the login method');
+    }
+
+    const body = new FormData();
+    body.append('routes', getReadStream());
+
+    return this.fetchRetryOnErrors<RoutesResponse>(
+      this.config.urls.uploadRoutesUrl,
+      {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.token}`,
+        },
+        body,
+      },
+      RETRY_ERRORS
+    ).catch(err => {
+      debugLog(`Something went wrong while trying to upload the routes`, err);
+      throw new ErrorUploadRoutes(err.message);
+    });
+  }
+
+  public async uploadContent(
+    getReadStream: () => fs.ReadStream
+  ): Promise<ContentResponse> {
+    if (!this.token) {
+      throw new Error(
+        'Before uploading content you must call the login method'
+      );
+    }
+
+    const body = new FormData();
+    body.append('importfile', getReadStream());
+
+    return this.fetchRetryOnErrors<ContentResponse>(
+      this.config.urls.uploadContentUrl,
+      {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.token}`,
+        },
+        body,
+      },
+      RETRY_ERRORS,
+      true
+    )
+      .then((res: ContentResponse) => {
+        if (res.problems && res.problems.length) {
+          res.problems.forEach(p =>
+            debugLog(`Content Warning: ${p.message}`, p.context)
+          );
         }
 
         return res;
       })
       .catch(err => {
-        console.log(
-          `Something went wrong while trying to uplaod the theme`,
-          err
+        debugLog(
+          `Something went wrong while trying to upload the content`,
+          err.message
         );
-        throw new ErrorUploadTheme();
+        throw new ErrorUploadContent(err.message);
       });
   }
 
+  private sleep(ms: number): Promise<void> {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  /**
+   * @todo this doesn't work as for some reason the retried fetch times out
+   */
+  private async fetchRetryOnErrors<R extends GhostResponse>(
+    url: string | Request,
+    init: RequestInit = { method: 'GET' },
+    retryErrorTypes: string[],
+    verifyStatusWithConfigFetch: boolean = false
+  ): Promise<R> {
+    const makeAttempt = (retries: number): Promise<R> => {
+      debugLog(`Calling ${init.method} ${url}`);
+      return fetch(url, {
+        ...init,
+        ...{ timeout: getenv.int('GHOST_API_FETCH_TIMEOUT_MS', 10000) },
+      })
+        .then(res => res.json() as Promise<R>)
+        .then((res: R) => {
+          if (res.errors) {
+            if (res.errors.some(e => retryErrorTypes.includes(e.errorType))) {
+              const message = `${init.method} ${url} failed due to ${
+                res.errors.find(e => retryErrorTypes.includes(e.errorType))!
+                  .errorType
+              }`;
```

Comment:
`.find` will only display one whereas here we might have multiple ones (can probably use `ghostErrorResponseToString`)